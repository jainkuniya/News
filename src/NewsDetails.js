//Import libraries
import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import styles from './styles'

//more styles
const moreStyles = StyleSheet.create({
	containerStyle: {
		marginTop: 16,
	},
});

//create component
export default class NewsDetails extends Component{
	

	render(){
		const { news } = this.props;
		const { author,
				description,
				publishedAt,
				title,
				url,
				urlToImage } = news;

		return (
			<View style={[styles.cardView, moreStyles.containerStyle]}>
				<Text>{title}</Text>
			</View>
		);
	}
}
