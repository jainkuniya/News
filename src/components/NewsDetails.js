//Import libraries
import React, { Component } from 'react';
import { View, TouchableOpacity, Text, StyleSheet, Image } from 'react-native';
import styles from '../common/styles';

//more styles
const moreStyles = StyleSheet.create({
	containerStyle: {
		marginTop: 12,
		flex: 1,
	},
	imageStyle: {
		height: 200,   
	},
	titleStyle: {
		fontSize: 14,
		color: 'black',
		position: 'absolute',
		backgroundColor: 'rgba(255,255,255,0.6)',
		bottom: 0,
		padding: 8,
	},
	descriptionStyle: {
		padding: 8,
	},
	authorStyle: {
		fontSize: 14,
		color: 'brown',
		position: 'absolute',
		backgroundColor: 'rgba(255,255,255,0.8)',
		top: 0,
		padding: 2,
	},
	publishedAtStyle: {
		fontSize: 10,
		justifyContent: 'flex-start',
		paddingBottom: 2,
	},
});

//create component
export default class NewsDetails extends Component {
	

	render() {
		const { news, openBroweser } = this.props;
		const { 
			author,
			description,
			publishedAt,
			title,
			urlToImage 
		} = news;
		const { 
			containerStyle,
			imageStyle,
			titleStyle,
			descriptionStyle,
			authorStyle,
			publishedAtStyle,
		} = moreStyles;
		console.log(urlToImage);
		return (
			<TouchableOpacity 
				style={[styles.cardView, containerStyle]}
				activeOpacity={0.7}
				onPress={openBroweser}
			>
				<View style={{ alignSelf: 'stretch' }}>
					<Image
						style={imageStyle}
						source={{ uri: urlToImage }}
					/>
					<Text style={authorStyle}>{author}</Text>
					<Text style={titleStyle}>{title}</Text>
				</View>
				<Text style={descriptionStyle}>{description}</Text>
				<Text style={publishedAtStyle}>{publishedAt}</Text>
			</TouchableOpacity>
		);
	}
}
