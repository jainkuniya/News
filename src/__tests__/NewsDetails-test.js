import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import NewsDetails from '../components/NewsDetails';

test('renders correctly', () => {
	const news = { 
		author: 'newsAuthor', 
		description: 'newsDescription', 
		publishedAt: 'newsTimeStamp', 
		title: 'newsTitle', 
		urlToImage: 'newsImageURL', 
	};
	const tree = renderer.create(
		<NewsDetails
			news={news} 
		/>
		).toJSON();
	expect(tree).toMatchSnapshot();
});
