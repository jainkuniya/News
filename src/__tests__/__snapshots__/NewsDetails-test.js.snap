// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "elevation": 2,
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 12,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "newsImageURL",
        }
      }
      style={
        Object {
          "height": 200,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.8)",
          "color": "brown",
          "fontSize": 14,
          "padding": 2,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      newsAuthor
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.6)",
          "bottom": 0,
          "color": "black",
          "fontSize": 14,
          "padding": 8,
          "position": "absolute",
        }
      }
    >
      newsTitle
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "padding": 8,
      }
    }
  >
    newsDescription
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 10,
        "justifyContent": "flex-start",
        "paddingBottom": 2,
      }
    }
  >
    newsTimeStamp
  </Text>
</View>
`;
